format/view-spec.md

```diff
@@ -55,211 +56,275 @@ Writers create view metadata files optimistically, assuming that the current met
 
-| Required/Optional | Field Name | Description |
-|-------------------|------------|-------------|
-| Required | timestamp-ms | The timestamp when the referenced version was made the current version |
-| Required | version-id | Version id of the view  |
+| Requirement | Key              | Value |
+|-------------|------------------|-------|
+| _required_  | `operation`      | Operation that caused this metadata to be created; must be `create` or `replace` |
+| _optional_  | `engine-name`    | Name of the engine that created the view version |
+| _optional_  | `engine-version` | Version of the engine that created the view version |
 
-#### Summary
+#### Representations
 
-Field “summary” is a string map with the following keys. Only `operation` is required. Engines may store additional key-value pairs in this map.
+View definitions can be represented in multiple ways. Representations are documented ways to express a view definition.
 
-| Required/Optional | Key | Value |
-|-------------------|-----|-------|
-| Required | operation | A string value indicating the view operation that caused this metadata to be created. Allowed values are “create” and “replace”. |
-| Optional | engine-version | A string value indicating the version of the engine that performed the operation |
+A view version can have more than one representation. All representations for a version must express the same underlying definition. Engines are free to choose the representation to use.
 
-#### Representations
+Each representation is an object with at least one common field, `type`, that is one of the following:
+* `sql`: a SQL SELECT statement that defines the view
+
+Representations further define metadata for each type.
+
+##### SQL representation
+
+The SQL representation stores the view definition as a SQL SELECT, with metadata such as the SQL dialect.
+
+| Requirement | Field name          | Type           | Description |
+|-------------|---------------------|----------------|-------------|
+| _required_  | `type`              | `string`       | Must be `sql` |
+| _required_  | `sql`               | `string`       | A SQL SELECT statement |
+| _required_  | `dialect`           | `string`       | The dialect of the `sql` SELECT statement (e.g., "trino" or "spark") |
+| _optional_  | `schema-id`         | `int`          | ID of the schema produced by the SELECT statement |
+| _optional_  | `default-catalog`   | `string`       | Catalog name to use when a reference in the SELECT does not contain a catalog |
+| _optional_  | `default-namespace` | `list<string>` | Namespace to use when a reference in the SELECT is a single identifier |
+| _optional_  | `field-aliases`     | `list<string>` | Column names optionally specified in the create statement |
+| _optional_  | `field-docs`        | `list<string>` | Column descriptions (COMMENT) optionally specified in the create statement |
```

Comment:
@amogh-jahagirdar, this is currently `field-comments` in the implementation. Which one do you want to change?

Reply:
Good catch, I think I'd prefer to change the spec to be `field-comments`. My rationale is that most operations (at least via SQL) will actually explicitly refer to the word "comment" so it seems more natural. 

for example `CREATE VIEW v (alias_name COMMENT 'docs', alias_name2, ...) AS SELECT col1, col2, ...`

Reply:
Updated. Please have another look and I'll commit when you +1.